BUILD

```diff
@@ -5272,46 +5272,14 @@ grpc_cc_library(
     ],
 )
 
-grpc_cc_library(
+grpc_upb_proto_library(
     name = "proto_gen_validate_upb",
-    srcs = [
-        "src/core/ext/upb-generated/validate/validate.upb.c",
-    ],
-    hdrs = [
-        "src/core/ext/upb-generated/validate/validate.upb.h",
-    ],
-    external_deps = [
-        "upb_lib",
-        "upb_lib_descriptor",
-        "upb_generated_code_support__only_for_generated_code_do_not_use__i_give_permission_to_break_me",
-    ],
-    language = "c++",
-    deps = [
-        "protobuf_duration_upb",
-        "protobuf_timestamp_upb",
-    ],
+    deps = ["@com_envoyproxy_protoc_gen_validate//validate:validate_proto"],
 )
 
-grpc_cc_library(
+grpc_upb_proto_reflection_library(
     name = "proto_gen_validate_upbdefs",
-    srcs = [
-        "src/core/ext/upbdefs-generated/validate/validate.upbdefs.c",
-    ],
-    hdrs = [
-        "src/core/ext/upbdefs-generated/validate/validate.upbdefs.h",
-    ],
-    external_deps = [
-        "upb_lib",
-        "upb_lib_descriptor_reflection",
-        "upb_textformat_lib",
-        "upb_reflection",
-        "upb_generated_code_support__only_for_generated_code_do_not_use__i_give_permission_to_break_me",
-    ],
-    language = "c++",
-    deps = [
-        "proto_gen_validate_upb",
-        "protobuf_timestamp_upbdefs",
-    ],
+    deps = ["@com_envoyproxy_protoc_gen_validate//validate:validate_proto"],
 )
```

Comment:
optional: Reading the internal code, both `grpc_upb_proto_library` and `grpc_upb_proto_reflection_library` shared the same deps for OSS and google3 build. It might be possible to reduce the number of targets.

Reply:
Both share the same deps but how can we reduce the number of targets?

Reply:
We might be able to exclude upb target from split into oss/ and google/.

Reply:
Yep, it sounds worth trying it. Thanks for bringing this up!